diag_manager/fms_diag_yaml.F90

```diff
@@ -182,13 +233,19 @@ subroutine fill_in_diag_files(diag_yaml_id, diag_file_id, fileobj)
     call diag_get_value_from_key(diag_yaml_id, sub_region_id(1), "grid_type", fileobj%file_sub_region%grid_type)
     if (trim(fileobj%file_sub_region%grid_type) .eq. "latlon") then
       allocate(fileobj%file_sub_region%lat_lon_sub_region(8))
+      fileobj%file_sub_region%lat_lon_sub_region = -999.
       call get_sub_region(diag_yaml_id, sub_region_id(1), fileobj%file_sub_region%lat_lon_sub_region)
     elseif (trim(fileobj%file_sub_region%grid_type) .eq. "index") then
       allocate(fileobj%file_sub_region%index_sub_region(8))
+      fileobj%file_sub_region%index_sub_region = -999
```

Comment:
Use [DIAG_NULL](https://github.com/NOAA-GFDL/FMS/blob/dmUpdate/diag_manager/diag_data.F90#L68) instead of hard coded value.